### PR TITLE
Only use signal channel if running embedded

### PR DESCRIPTION
**Pull request by arnikola**
Fixes an issue where m3 can hang if stopped during initialization

### CODE REVIEW

File: src/query/server/server.go [hunk 1]
```diff
@@ -196,17 +196,25 @@ func Run(runOpts RunOptions) {
...
-	case <-sigChan:
-	case <-interruptCh:
+	if cfg.Backend == config.GRPCStorageType {
+		// Only use this if not running embedded, otherwise will
+		// obfuscate signal channel for m3db
+		sigChan := make(chan os.Signal, 1)
+		signal.Notify(sigChan, os.Interrupt, syscall.SIGTERM)
+
+		select {
+		case <-sigChan:
```
Comment (prateek) [inline src/query/server/server.go]: nit: log the cause of termination so you can track why the process crashed

```
var interruptErr error
select {
case sig := <- sigChan:
  interruptErr = fmt.Errorf("%v", sig)
case err  := <- interruptCh:
  interruptErr = err
}

log.Infof("interrupt :%v", interruptErr)

Comment (arnikola) [reply src/query/server/server.go]: Good call

**Comment by prateek** [comment]
@arnikola looks like metalint is :(

**Comment by arnikola** [comment]
Unfortunately a repro test would be super messy; better to add this as an integration test when we have those up and running :)

**Pull request merged**